initial working MVP

MichaelDimmitt opened the pull request.

kevinramharak commented:
@MichaelDimmitt I think this is good workaround to deliver the feature in a working state, but it does circumvent the additions in the `package.json`.

The addition of:
```json
      {
          "language": "type-markdown-injection",
          "scopeName": "markdown.type.codeblock",
          "path": "./syntaxes/type-markdown-injection.tmGrammar.json",
          "injectTo": [
              "text.html.markdown"
          ],
          "embeddedLanguages": {
              "meta.embedded.block.type": "type"
          }
      }
```
Is supposed to register <code>```type<code> as a code block where the highlighting is based on the `type-markdown-injection.tmGrammar.json` definition. I could not figure out why it doesn't get used, even when using examples copied verbatim. Its a bit niche, and hard to debug. So I don't expect to have the time to figure it out. If you have the time, maybe give it a try.

If not I think this is a good alternative, that I had not considered.

MichaelDimmitt commented:
I looked at another plugin
https://github.com/mjbvz/vscode-github-markdown-preview - which led me to:
https://github.com/mjbvz/vscode-markdown-emoji/blob/master/notebook/index.ts#L16

It looks like vscode-markdown-emoji re-writes the markdown using the following packages.
https://github.com/markdown-it/markdown-it-emoji
https://github.com/markdown-it/markdown-it

MichaelDimmitt commented:
Sorry for the book of info. 📚
Curious to hear any thoughts.

MichaelDimmitt commented:
@kevinramharak , 

I noticed the markdown previews only matched 80% with the tooltip that was showing.
Library Types such as ZodDescriminatedUnion were showing as a different color.

So I added a commit in this pr which introduced a css file that overrides the css class that is generated by the markdown preview. In that css file I changed the color to the one showing in the preview.

Do you know if a change will work like this will work when people change vscode color themes?

## Screenshots
Here are some screenshots showing old and new with `Developer: Open Webview Developer Tools` inspecting the dom element styles.

## Old
<img width="1512" alt="Screenshot 2025-06-30 at 1 26 00 PM" src="https://github.com/user-attachments/assets/3728c41f-c344-4200-8dec-51e9bf0274ef" />

## New
<img width="1512" alt="Screenshot 2025-06-30 at 1 28 40 PM" src="https://github.com/user-attachments/assets/722232f7-e879-4bad-b305-539abc5a93e9" />

MichaelDimmitt commented:
@kevinramharak @yoavbls 
Is there any interest in this pr?

Happy to make changes. 
Please let me know if anything is blocking or needs to get in before the feature is in a mergable state.

Cheers, 
Michael Dimmitt

yoavbls commented:
@MichaelDimmitt BTW, I published the lib I talked about:
https://github.com/yoavbls/vscode-shiki-bridge

Would you like to collaborate on that?

kevinramharak commented on src/components/title.ts:
```diff
@@ -2,15 +2,18 @@ import { Diagnostic } from "vscode-languageserver-types";
-            ${errorCodeExplanationLink(diagnostic.code)}  | 
+            (TS${diagnostic.code})
+            ${errorCodeExplanationLink(diagnostic.code)}  |
             ${errorMessageTranslationLink(diagnostic.message)}
```
This is because the `codeicon` icon font is not loaded.

kevinramharak commented:
@yoavbls @MichaelDimmitt I reimplemented this code in a new branch, apologies for losing the commits here, but I did not want to deal with solving the conflicts of merging in the monorepo changes, as its easy to forget something (as shown in #51).

I think I summarized every discussion point here, but feel free to add more at #152.
